core/trino-main/src/main/java/io/trino/operator/project/PageProcessor.java

```diff
@@ -328,9 +342,11 @@ private ProcessBatchResult processBatch(int batchSize)
                 }
                 else {
                     if (pageProjectWork == null) {
+                        Page inputPage = projection.getInputChannels().getInputChannels(page);
                         expressionProfiler.start();
-                        pageProjectWork = projection.project(session, yieldSignal, projection.getInputChannels().getInputChannels(page), positionsBatch);
-                        expressionProfiler.stop(positionsBatch.size());
+                        pageProjectWork = projection.project(session, yieldSignal, inputPage, positionsBatch);
+                        long projectionTimeNanos = expressionProfiler.stop(positionsBatch.size());
+                        metrics.recordProjectionTime(projectionTimeNanos);
```

Comment:
@raunaqmorarka FYI I think this does not measure projections time, `pageProjectWork.process()` is the bulk of work here

Reply:
You are right, `pageProjectWork.process()` does the stuff

Reply:
yes, I assumed that the `expressionProfiler` was in the right place, that also needs to be around `pageProjectWork.process()`

Reply:
Created https://github.com/trinodb/trino/pull/15364